File: javascript/bob/bob.js
var shouting = /^[^a-z]+$/;
var silence  = /^\s*$/;
var question = /\?$/;

var Bob = function() {};

Bob.prototype.hey = function(message) {
  switch (true) {
    case silence.test(message):
      return 'Fine. Be that way!';
    case shouting.test(message):
      return 'Woah, chill out!';
    case question.test(message):
      return 'Sure.';
    default: 
      return 'Whatever.';
  }
};

module.exports = Bob;

File: elixir/bob/bob.exs
defmodule Teenager do
  def hey(message) do
    cond do
      silence?(message)  -> "Fine. Be that way!"
      question?(message) -> "Sure."
      shouting?(message) -> "Woah, chill out!"
      true               -> "Whatever."
    end
  end

  defp question?(message) do
    String.ends_with?(message, "?")
  end

  defp shouting?(message) do
    message == String.upcase(message)
  end

  defp silence?(message) do
    String.strip(message) == ""
  end
end
